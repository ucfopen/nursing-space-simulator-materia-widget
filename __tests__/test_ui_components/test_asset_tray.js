import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import AssetTray from "../../src/js/components/ui/asset_tray";

// Constants used across all tests
const setCategoryMock = jest.fn();
const selectAssetTypeMock = jest.fn();

const assets = {
	bed: {
		category: "beds"
	},
	chair: {
		category: "equipment"
	}
};
const currentCategory = "construction";
const categories = ["beds", "equipment", "walls", "people"];
const isMenuVisible = true;
const selectedAsset = { id: "pov_camera" };

describe("Asset Tray tests", () => {
	it("renders asset tray", () => {
		const assetTrayTree = renderer
			.create(
				<AssetTray
					assets={assets}
					categories={categories}
					currentCategory={currentCategory}
					setCategory={setCategoryMock}
					selectAssetType={selectAssetTypeMock}
					selectedAsset={selectedAsset}
				/>
			)
			.toJSON();

		expect(assetTrayTree).toMatchSnapshot();
	});

	it("toggles drawer correctly", () => {
		const mockToggleMenu = jest.fn();
		const assetTray = shallow(
			<AssetTray
				assets={assets}
				categories={categories}
				currentCategory={currentCategory}
				setCategory={setCategoryMock}
				selectAssetType={selectAssetTypeMock}
				isMenuVisible={isMenuVisible}
				toggleMenu={mockToggleMenu}
			/>
		);

		expect(assetTray.find(".drawer-toggle").text()).toEqual("[Close Menu]");
		assetTray.find(".drawer-toggle").simulate("click");
		expect(mockToggleMenu).toHaveBeenCalled();
		assetTray
			.find("#UI-bottom-panel")
			.simulate("mousedown", { preventDefault: () => {} });
	});

	it("selects POV Camera correctly", () => {
		const assetTray = shallow(
			<AssetTray
				assets={assets}
				categories={categories}
				currentCategory={currentCategory}
				setCategory={setCategoryMock}
				selectAssetType={selectAssetTypeMock}
			/>
		);

		assetTray
			.find("button")
			.at(1)
			.simulate("click");

		expect(selectAssetTypeMock).toBeCalledWith({
			id: "pov_camera",
			title: "POV Camera"
		});
	});
});
