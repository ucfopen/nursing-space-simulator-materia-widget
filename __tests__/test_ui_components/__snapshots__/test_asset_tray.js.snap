// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Tray tests renders asset tray 1`] = `
<div
  className="open"
  id="UI-bottom-panel"
>
  <button
    className="drawer-toggle"
    onClick={[Function]}
  >
    [Close Menu]
  </button>
  <div
    id="asset-selection-menu"
  >
    <button
      id="vr-viewer-mode"
      onClick={[Function]}
    >
      First-Person Viewer
    </button>
    <div
      id="categories-list"
    >
      <button
        className="asset-category"
        data-category="beds"
        onClick={[Function]}
      >
        beds
      </button>
      <button
        className="asset-category"
        data-category="equipment"
        onClick={[Function]}
      >
        equipment
      </button>
      <button
        className="asset-category"
        data-category="walls"
        onClick={[Function]}
      >
        walls
      </button>
      <button
        className="asset-category"
        data-category="people"
        onClick={[Function]}
      >
        people
      </button>
    </div>
  </div>
  <div
    id="asset-picker"
  >
    <div
      className="button-container"
    >
      <button
        className="asset"
        data-category="beds"
        data-title={undefined}
        id={undefined}
        onClick={[Function]}
        style={
          Object {
            "background": "url(undefined) no-repeat center center",
            "backgroundSize": "100% 100%",
          }
        }
      />
      <div />
    </div>
  </div>
</div>
`;
